src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,331 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    public class KDCSetup
+    {
+        private const string Krb5ConfigFile = "/etc/krb5.conf";
+        private const string KDestroyCmd = "kdestroy";
+
+        public KDCSetup()
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+            startInfo.UseShellExecute = true;
+            startInfo.FileName = "./setup-kdc.sh";
+            Process kdcSetup = new Process();
+            kdcSetup.StartInfo = startInfo;
+            kdcSetup.Start();
+            kdcSetup.WaitForExit();
+            Assert.Equal(0, kdcSetup.ExitCode);
+        }
+
+        // checks for avilability of Kerberos related infrastructure
+        // on the host. Returns true available, false otherwise
+        public static bool CheckAndInitializeKerberos()
+        {
+            if (File.Exists(Krb5ConfigFile))
+            {
+                // Clear the credentials
+                var startInfo = new ProcessStartInfo(KDestroyCmd);
+                startInfo.UseShellExecute = false;
+                startInfo.CreateNoWindow = true;
+                startInfo.Arguments = "-A";
+                Process clearCreds = Process.Start(startInfo);
+                clearCreds.WaitForExit();
+                return (clearCreds.ExitCode == 0);
+            }
+            return false;
+        }
+    }
+
+    public class KerberosTest : IDisposable, IClassFixture<KDCSetup>
+    {
+        private readonly byte[] _sampleMsg = Encoding.UTF8.GetBytes("Sample Test Message");
+        private readonly bool _isKrbAvailable; // tests are no-op if kerberos is not available on the host machine
+        private readonly KDCSetup _fixture;
+
+        public KerberosTest(KDCSetup fixture)
+        {
+            _fixture = fixture;
+            _isKrbAvailable = KDCSetup.CheckAndInitializeKerberos();
```

Comment:
> My understanding is that this is going to run before each individual test... is that desired (rather than once before all tests)?

Setup happens once for all the tests.

Before each test, "kdestroy -A" is run. A single instance of KDCSetup is passed to all the tests.